Add Lightning Network integration documentation (#90)
arkade-os/go-sdk: docs/lightning.md

|-----------|-----------|-----------|--------------|
| **Receive from Lightning** | Reverse submarine swap | Nothing (payer sends LN payment) | VHTLC in Arkade |
| **Send to Lightning** | Submarine swap | VHTLC in Arkade | Lightning invoice paid |

Both directions use a **VHTLC** (Virtual Hash Time-Lock Contract) as the atomic
swap primitive, ensuring that either the swap completes in full or both parties
can recover their funds.

---

## Key Concepts

### Virtual HTLC (VHTLC)

A VHTLC is an off-chain HTLC that lives inside the Ark protocol's virtual
transaction tree. It is composed of multiple Tapscript leaves that encode
different spending conditions:

| Leaf | Spending Condition | Purpose |
|------|-------------------|---------|
| **Claim** | Receiver + Server + preimage | Normal claim by the receiver |
| **Refund** | Sender + Server + timelock | Cooperative refund after timeout |
| **Unilateral Claim** | Receiver + preimage + timelock | Fallback claim without server |
| **Unilateral Refund** | Sender + timelock | Fallback refund without server |
| **Unilateral Refund Without Receiver** | Sender + longer timelock | Last-resort refund without receiver or server |

> The server (Ark Service Provider, or ASP) is a co-signer on the cooperative

Reviewer: ```suggestion
The Arkade operator is a co-signer on the cooperative
```

    RefundPublicKey: hex.EncodeToString(refundPubKey),
})

// 3. Send funds to the swap address
receivers := []types.Receiver{
    {To: submarineSwap.Address, Amount: submarineSwap.ExpectedAmount},
}
txid, err := arkClient.SendOffChain(ctx, receivers)

// 4. Monitor status via Boltz WebSocket or polling
// Wait for "transaction.claimed" status

// 5. Retrieve preimage as proof of payment
preimage := boltzClient.GetSwapPreimage(submarineSwap.ID)
```

---

## Refunding Stuck Payments

A submarine swap (send to Lightning) can get **stuck** if:
- The Lightning invoice expires before Boltz can pay it
- Boltz fails to route the payment
- The swap times out
- A network error interrupts the process

In these cases, your funds are locked in a VHTLC at the swap address. You need
to execute a **refund** to recover them.

### When Is a Refund Needed?

Monitor the swap status. These statuses indicate the swap failed and a refund is
possible:

| Status | Meaning |
|--------|---------|
| `swap.expired` | The swap timed out |
| `invoice.failedToPay` | Boltz could not pay the Lightning invoice |
| `transaction.lockupFailed` | The lockup transaction failed |

### Refund Flow

```

> Your Wallet                Boltz              Ark Server (ASP)

Reviewer: ```suggestion
Your Wallet                Boltz              Arkade Server
```

1. **Detect the failure**: Poll the swap status or listen for WebSocket
   updates. If the status is `swap.expired`, `invoice.failedToPay`, or
   `transaction.lockupFailed`, a refund is needed.

2. **Locate the VHTLC**: Query the indexer for VTXOs at the swap address.
   These are the locked funds to recover.

3. **Build the refund transaction**: Construct an off-chain transaction that
   spends the VHTLC using the **refund** leaf:
   - If the VTXO is **not recoverable** (still in the virtual tree): use the
     cooperative refund path (sender + Boltz + server). Request Boltz's
     co-signature via `POST /v2/swap/submarine/{id}/refund`, then submit to
     the ASP.
   - If the VTXO is **recoverable** (has been swept on-chain): use the
     `refundWithoutReceiver` leaf and join a batch via `RegisterIntent`.

4. **Get Boltz co-signature** (non-recoverable path): Send the unsigned
   refund transaction and checkpoint transaction to Boltz. Boltz signs its
   portion and returns the partially signed transactions.

5. **Sign your portion**: Sign the refund and checkpoint transactions with
   your wallet key.

6. **Combine signatures**: Merge the Boltz signatures with yours.

> 7. **Submit to the ASP**: Call `SubmitTx` with the combined refund

Reviewer: ```suggestion
7. **Submit to Arkade**: Call `SubmitTx` with the combined refund
```

   back to your Arkade wallet address.

### Go SDK Integration Pattern

```go
// 1. Detect failed swap
status := boltzClient.GetSwapStatus(swapID)
if status == "swap.expired" || status == "invoice.failedToPay" {
    // Refund is needed
}

// 2. Query the VHTLC VTXO from the indexer
vtxos, err := indexer.GetVtxos(ctx, swapAddressScript)

// 3. Build the VHTLC refund leaf script
// The refund leaf requires: sender (you) + Boltz + server signatures
refundLeaf := buildVHTLCRefundLeaf(yourPubKey, boltzPubKey, serverPubKey, timeout)

// 4. Build the off-chain refund transaction
refundTx, checkpointTxs := buildOffchainTx(vtxoInput, yourOutput, checkpointScript)

// 5. Request Boltz co-signature
boltzSigned := boltzClient.RefundSubmarineSwap(swapID, refundTx, checkpointTxs[0])

// 6. Sign your part
yourSignedRefundTx, _ := arkClient.SignTransaction(ctx, refundTx)
yourSignedCheckpoint, _ := arkClient.SignTransaction(ctx, checkpointTxs[0])

// 7. Combine signatures
combinedRefundTx := combineTapscriptSigs(boltzSigned.RefundTx, yourSignedRefundTx)
combinedCheckpoint := combineTapscriptSigs(boltzSigned.Checkpoint, yourSignedCheckpoint)

> // 8. Submit to ASP

Reviewer: ```suggestion
// 8. Submit
```
